nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -429,8 +429,7 @@ public void testForMethodReferenceInAnAssignment() {
             "    return o.toString();",
             "  }",
             "  static void testPositive() {",
-            "    // TODO: we should report an error here, since Test::foo cannot take",
-            "    // a @Nullable parameter.  we don't catch this yet",
+            "    // BUG: Diagnostic contains: referenced method is @NonNull",
```

Comment:
Right now, the full error message we emit here is:
```
warning: [NullAway] parameter o of referenced method is @NonNull, but parameter in functional interface method com.uber.Test.A.function(T1) is @Nullable
```
It'd be nice if we could find a way to print the functional interface method type so it refers to the annotated type arguments rather than the `T1` type variable.

Reply:
I agree, I'll try and find a way to do that

Reply:
After further investigation getting better error messages needs more thought.  I'll file a follow-up issue but let's not hold up this PR on this.